torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -329,37 +339,47 @@ def from_hp_to_intx(
             )
             data = data.to(target_dtype)
         else:
-            scale, zero_point = choose_qparams_affine(
-                input_float,
-                mapping_type,
-                block_size,
-                target_dtype,
-                quant_min,
-                quant_max,
-                eps,
-                scale_dtype,
-                zero_point_dtype,
-                preserve_zero,
-                zero_point_domain,
-            )
-            # choose_qparams_affine is a custom op that does support returning optional Tensors. We thus set the zero_point to None if its domain is None
-            if zero_point_domain is None:
-                zero_point = None
-            data = quantize_affine(
-                input_float,
-                block_size,
-                scale,
-                zero_point,
-                target_dtype,
-                quant_min,
-                quant_max,
-                zero_point_domain,
-            )
-            # Note: output will be uint8 tensor for sub byte tensors for now
+            if isinstance(_layout, MarlinQQQLayout):
```

Comment:
@jerryzh168 

All this if else branching makes me feel like something is wrong with the abstraction

Reply:
yeah I don't think this is intended actually, AffineQuantizedTensor is corresponding to a very specific quantize/dequantize algorithm, so if there is a different quantize/dequantize people will need to create another tensor subclass

Reply:
@jerryzh168 make sense.